pkg/proxy/endpoints.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+// EndpointsChangeMap maps a service to its endpointsChange.
+type EndpointsChangeMap struct {
+	// lock protects items.
+	lock sync.Mutex
+	// hostname is the host where kube-proxy is running.
+	hostname string
+	// items is a service to its endpointsChange map.
+	items map[types.NamespacedName]*endpointsChange
```

Comment:
To me, it is cleaner if the comment reads something like: "items maps a service to an endpointsChange". If you wanted to be thorough, you could even add a little more. Possibly something like "this is used to keep track of what endpoints changed for each service"

Reply:
Sounds fair. It's fixed now.